### PR TITLE
Include missing visitors and type checking for on clause

### PR DESCRIPTION
**Pull request by lasinicl**
## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/OnClauseTest.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 /**
  * This contains methods to test on clause in query expression.
  *
+ *
```
Comment (grainier) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/query/OnClauseTest.java]: Remove additional space.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/references/SymbolReferenceFindingVisitor.java [hunk 2]
```diff
@@ -920,6 +921,13 @@ public void visit(BLangWhereClause whereClause) {
         this.acceptNode(whereClause.expression);
     }
 
+    @Override
+    public void visit(BLangOnClause onClause) {
+        this.acceptNode(onClause.expression);
+    }
+
+
```
Comment (grainier) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/util/references/SymbolReferenceFindingVisitor.java]: Remove additional space.

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 3]
```diff
@@ -3298,7 +3299,9 @@ public void visit(BLangQueryExpr queryExpr) {
             } else if (kind == NodeKind.ON_CONFLICT) {
                 typeCheckOnConflictClause((BLangOnConflictClause) clause, parentEnv);
             } else if (kind == NodeKind.WHERE) {
-                whereEnv = typeCheckWhereClause((BLangWhereClause) clause, select, parentEnv);
+                whereEnv = typeCheckWhereClause(((BLangWhereClause) clause).expression, select, parentEnv);
+            }else if (kind == NodeKind.ON) {
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:3303]: Please format the added change

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java [hunk 4]
```diff
@@ -3407,7 +3410,9 @@ public void visit(BLangQueryAction queryAction) {
             } else if (kind == NodeKind.LET_CLAUSE) {
                 parentEnv = typeCheckLetClause((BLangLetClause) clause, parentEnv);
             } else if (kind == NodeKind.WHERE) {
-                whereEnv = typeCheckWhereClause((BLangWhereClause) clause, doClause, parentEnv);
+                whereEnv = typeCheckWhereClause(((BLangWhereClause) clause).expression, doClause, parentEnv);
```
Comment (pcnfernando) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:3413]: Not formated here as well

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java (hunk 3 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:3303]: Formatting issues have been rectified in https://github.com/ballerina-platform/ballerina-lang/pull/23252

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java (hunk 4 above)
Comment (lasinicl) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java:3413]: Formatting issues have been rectified in https://github.com/ballerina-platform/ballerina-lang/pull/23252

**Pull request merged**